src/bootstrap/builder.rs

```diff
@@ -939,24 +939,6 @@ impl<'a> Builder<'a> {
 
         if !mode.is_tool() {
             cargo.env("RUSTC_FORCE_UNSTABLE", "1");
-
-            // Currently the compiler depends on crates from crates.io, and
-            // then other crates can depend on the compiler (e.g., proc-macro
-            // crates). Let's say, for example that rustc itself depends on the
-            // bitflags crate. If an external crate then depends on the
-            // bitflags crate as well, we need to make sure they don't
-            // conflict, even if they pick the same version of bitflags. We'll
-            // want to make sure that e.g., a plugin and rustc each get their
-            // own copy of bitflags.
-
-            // Cargo ensures that this works in general through the -C metadata
-            // flag. This flag will frob the symbols in the binary to make sure
-            // they're different, even though the source code is the exact
-            // same. To solve this problem for the compiler we extend Cargo's
-            // already-passed -C metadata flag with our own. Our rustc.rs
-            // wrapper around the actual rustc will detect -C metadata being
-            // passed and frob it with this extra string we're passing in.
-            cargo.env("RUSTC_METADATA_SUFFIX", "rustc");
```

Comment:
Hm, so we probably want to still have a branch for __CARGO_DEFAULT_LIB_METADATA for rustc, right? Otherwise we'll run into conflicts between librustc's dependencies and tools (miri, clippy, etc) which are built in a separate lib dir. For some of them a possible solution is to just delete the dependencies from Cargo.toml like we did in codegen, but that won't work for e.g. Cargo which needs to build on stable.

Reply:
I'm honestly not 100% sure myself. This just sort of comes up randomly and we deal with it as it comes up, but given that the bootstrap proceeded and tools like rustdoc were built I don't think that we need to configure this for other tools/builds. If I'm wrong though it's easy enough to add back in!

Reply:
Definitely; if things work then we can leave it out.